pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -229,18 +220,22 @@ func detectAuthEnv(jwt string) (*authenticate.JwtPayload, error) {
 	return structuredPayload, nil
 }
 
-// loadRemoteCACerts mounts an existing cacerts Secret if the files aren't mounted locally.
+// loadCACerts loads an existing `cacerts` Secret if the files aren't mounted locally.
 // By default, a cacerts Secret would be mounted during pod startup due to the
 // Istiod Deployment configuration. But with external Istiod, we want to be
 // able to load cacerts from a remote cluster instead.
-func (s *Server) loadRemoteCACerts(caOpts *caOptions, dir string) error {
+func (s *Server) loadCACerts(caOpts *caOptions, dir string) error {
 	if s.kubeClient == nil {
 		return nil
 	}
 
 	signingKeyFile := path.Join(dir, ca.CAPrivateKeyFile)
-	if _, err := os.Stat(signingKeyFile); !os.IsNotExist(err) {
-		return fmt.Errorf("signing key file %s already exists", signingKeyFile)
+	if _, err := os.Stat(signingKeyFile); err == nil {
```

Comment:
Looking at this logic: long ago we were mounting the secrets in the pod, and Istiod was not watching secrets. We now watch secrets - the whole file path and delployment.yaml could be cleaned up. However - who knows what will break...

I think at some point we should start deprecating various code paths like this - and add some metric or status with all the obsolete/deprecated features/code paths.

Reply:
Thanks for the input. I can not agree more.